iface/get.go

```diff
@@ -121,6 +124,37 @@ func GetInterface(dir, ifaceName string) (Interface, error) {
 	return iface, nil
 }
 
+// explodeInterface traverses an interface type, returning the original
+// interface along with all transitively embedded interfaces.
+func explodeInterface(iface *types.Interface) []*types.Interface {
+	var (
+		result    []*types.Interface
+		workQueue = []*types.Interface{iface}
+		visited   = map[string]bool{}
+	)
+	for len(workQueue) > 0 {
+		current := workQueue[0]
+		workQueue = workQueue[1:]
+		currentID := current.String()
+		if !visited[currentID] {
+			visited[currentID] = true
+			result = append(result, current)
+			for i := 0; i < current.NumEmbeddeds(); i++ {
+				switch embedded := current.EmbeddedType(i).(type) {
+				case *types.Interface:
+					workQueue = append(workQueue, embedded)
+				case *types.Named:
+					switch underlying := embedded.Underlying().(type) {
+					case *types.Interface:
+						workQueue = append(workQueue, underlying)
+					}
+				}
```

Comment:
This should capture all the possibilities we care about. (And maybe everything that's possible?) Typically we only embed named interfaces, but this would also handle embedded anonymous interfaces, e.g.

```go
type A interface {
    interface { Foo() }
}
```